pmd-plsql/etc/grammar/PldocAST.jjt

```diff
@@ -3166,6 +3166,8 @@ ASTUnaryExpressionNotPlusMinus UnaryExpressionNotPlusMinus() #UnaryExpressionNot
 {
   (
     <NOT> {sb.append(" NOT "); } simpleNode = UnaryExpression(false)
+  |
+    LOOKAHEAD(2) simpleNode = DateTimeLiteral()
```

Comment:
We actually need to fix PrimaryPrefix - where the datetimeliteral should have been already covered by Literal(). However, since e.g. "TIMESTAMP" is not a reserved keyword, it was interpreted as a identifier.
I'll fix it in PimaryPrefix.
Thanks for the test case!